types/react/experimental.d.ts

```diff
@@ -38,7 +38,24 @@ import React = require('.');
 
 export {};
 
+type NonNullableValue = string | number | boolean | object | symbol;
+
 declare module '.' {
+    type MutableSource<Source extends NonNullableValue, Version extends NonNullableValue> = Pick<{
```

Comment:
Like what @eps1lon said, the `Pick<T, never>` will not work I'm pretty sure. I'm on board with his suggestions to create a branded key using a `unique symbol`. It doesn't allow access to it but maintains type inference.